include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -347,6 +347,32 @@ def FModuleLike : OpInterface<"FModuleLike", [Symbol, SymboledPortList, Instance
   }];
 }
 
+def ClassLike : OpInterface<"ClassLike", [FModuleLike]> {
```

Comment:
In context of narrowing interfaces a bit, does this being `FModuleLike` make sense?

(Probably doesn't support parameters, instantiation convention, or port symbols? -- looks like opts-out of parameters and hardcodes convention presently)

Not to be addressed when adding this, but just wondering if it should be module-like or if that's something we want to tease apart later?

Interestingly this also makes them `InnerSymbolTable`s which is may be right but is a bit surprising.

Reply:
Bit more thought, this might be more work presently than is worth -- esp for things like leveraging the insert/erase port helpers that manage all the arrays, so nevermind but something we can chat about.  They're not /unlike/ FModule's :smile: .